static/app/components/compactSelect/composite.tsx

```diff
@@ -0,0 +1,208 @@
+  extends BaseCompositeSelectRegion<Value>,
+    Omit<
+      MultipleListBoxProps<Value>,
+      'children' | 'items' | 'compositeIndex' | 'size' | 'onChange'
+    > {
+  onChange: (values: Value[]) => void;
+}
+
+/**
+ * A "region" inside a composite select. Each "region" is a separated, self-contained
+ * select box (each renders as a list box with its own list state) whose selection
+ * values don't interfere with one another.
+ */
+export type CompositeSelectRegion<Value extends React.Key> =
+  | SingleCompositeSelectRegion<Value>
+  | MultipleCompositeSelectRegion<Value>;
+
+/**
+ * [CompositeSelectChild description]
+ * @type {[type]}
+ */
+type CompositeSelectChild =
+  | React.ReactElement<CompositeSelectRegion<React.Key>>
+  | false
+  | null
+  | undefined;
+
+export interface CompositeSelectProps extends ControlProps {
+  /**
+   * The "regions" inside this composite selector. Each region functions as a separated,
+   * self-contained select box (each renders as a list box with its own list state)
+   * whose values don't interfere with one another.
+   */
+  children: CompositeSelectChild | CompositeSelectChild[];
+  /**
+   * Whether to close the menu upon selection. This prop applies to the entire selector
+   * and functions as a fallback value. Each composite region also accepts the same
+   * prop, which will take precedence over this one.
+   */
+  closeOnSelect?: SingleListBoxProps<React.Key>['closeOnSelect'];
+}
+
+/**
+ * Flexible select component with a customizable trigger button
+ */
+function CompositeSelect({
+  children,
+  // Control props
+  disabled,
+  size = 'md',
+  closeOnSelect,
+  ...controlProps
+}: CompositeSelectProps) {
+  return (
+    <Control {...controlProps} size={size} disabled={disabled}>
+      <FocusScope>
+        <RegionsWrap>
+          {Children.map(children, (child, index) => {
+            if (!child) {
+              return null;
+            }
+
+            return (
+              <Region
+                {...child.props}
+                size={size}
+                compositeIndex={index}
+                closeOnSelect={child.props.closeOnSelect ?? closeOnSelect}
+              />
+            );
+          })}
+        </RegionsWrap>
+      </FocusScope>
+    </Control>
+  );
+}
+
+/**
+ * A "region" inside composite selectors. Each "region" is a separated, self-contained
+ * select box (each renders as a list box with its own list state) whose selection
+ * values don't interfere with one another.
+ */
+CompositeSelect.Region = function <Value extends React.Key>(
+  _props: CompositeSelectRegion<Value>
+) {
+  // This pseudo-component not meant to be rendered. It only functions as a props vessel
+  // and composable child to `CompositeSelect`. `CompositeSelect` iterates over all child
+  // instances of `CompositeSelect.Region` and renders `Region` with the specified props.
+  return null;
+};
+
+export default CompositeSelect;
```

Comment:
Let's do `export {CompositeSelect}` I think we're trying to move away from default exports

Reply:
I'll open another PR to do `{CompactSelect}` also